From 4ac672c82ddf02ae3aa125043131d46f054956f0 Mon Sep 17 00:00:00 2001
From: Double Lo <dblo@cypress.com>
Date: Thu, 13 Sep 2018 03:07:29 -0500
Subject: [PATCH 007/100] non-upstream: avoid network disconnection during
 suspend with linux-3.8 and above kernel

From linux-3.8 kernel, wowlan packet filter is mandated to avoid
the disconnection of connected network before suspend in commit
812569699119 ("cfg80211/mac80211: disconnect on suspend").
So as a dummy wowlan filter is configured for kernels linux-3.8
and above. Later all private command implementation of packet
filters can be moved to wowlan based filters.

Signed-off-by: Double Lo <dblo@cypress.com>
Signed-off-by: Chi-hsien Lin <chi-hsien.lin@cypress.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index e3f47a9d4362..a165083e5d12 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -7168,8 +7168,7 @@ static int brcmf_setup_wiphy(struct wiphy *wiphy, struct brcmf_if *ifp)
 	wiphy->vendor_commands = brcmf_vendor_cmds;
 	wiphy->n_vendor_commands = BRCMF_VNDR_CMDS_LAST - 1;
 
-	if (brcmf_feat_is_enabled(ifp, BRCMF_FEAT_WOWL))
-		brcmf_wiphy_wowl_params(wiphy, ifp);
+	brcmf_wiphy_wowl_params(wiphy, ifp);
 	err = brcmf_fil_cmd_data_get(ifp, BRCMF_C_GET_BANDLIST, &bandlist,
 				     sizeof(bandlist));
 	if (err) {
-- 
2.17.1

